### DISCUSSION TITLE
How can i set the default rows to 10 per page with controlled pagination

### DISCUSSION THREAD

**Discussion by SandeepKapalawai** [TanStack/table#2875] [Q&A]
Hello Every one,

I started using the react table few days ago , i was trying to customize my table having a button which should should +10 more records, but up on initial loading i need to display  only 10 records per page.o 

I read the documentation even though if i didn't mention the pageSize it will take as default to 10, but what i see from my code it is defaulting to total number of available rows i.e, 14 which i was expecting 10 should be displayed at the first instance then later on when i click on the  show more button it should the other 4 records.

I tried the same example but am i missing something or is it abug,

Please find the below code sandbox

https://codesandbox.io/s/gifted-dream-xbqnd?file=/src/App.js

Thank you
@tannerlinsley

**Comment by ItzSudhan** [comment]
4 years and still nobody answered...

**Reply by KevinVandy** [reply]
Because it's multiple places in the docs

**Comment by OussamaKhoubran** [comment]
Use the `pageSize`

```
const table = useReactTable({
  columns,
  data,
  getCoreRowModel: getCoreRowModel(),
  getPaginationRowModel: getPaginationRowModel(),
  initialState: {
    pagination: {
      pageIndex: 2, //custom initial page index
      pageSize: 25, //custom default page size
    },
  },
});
```
